src/Rules/IssetCheck.php

```diff
@@ -60,8 +64,9 @@ public function check(Expr $expr, Scope $scope, string $operatorDescription, cal
 
 			return $error;
 		} elseif ($expr instanceof Node\Expr\ArrayDimFetch && $expr->dim !== null) {
-
-			$type = $scope->getType($expr->var);
+			$type = $this->treatPhpDocTypesAsCertain
+				? $scope->getType($expr->var)
+				: $scope->getNativeType($expr->var);
 			$dimType = $scope->getType($expr->dim);
```

Comment:
Can you please write a test that verifies `$dimType` should get the same treatment, and them commit a fix for `$dimType` as well? Thanks.

Reply:
Sure, but could you help me to write it/understand this code ?
I don't know what is `$dimType`.

Reply:
`$dimType` is the type of `ArrayDimFetch::$dim`. If you don't know what that is, please read this guide: https://phpstan.org/developing-extensions/abstract-syntax-tree

Reply:
I recommend phpast.com to get familiar with the AST from PHP-Parser

Reply:
it could be better though, it doesn't show the node class names :/

Reply:
Thanks ! `phpast.com` help me to understand that dim was the key of the array.
I added a test.

Reply:
> it could be better though, it doesn't show the node class names :/

Yeah, it can be reconstructed with nodeType, but it could be great to have the full class directly